runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/SimpleMonitoringInfoBuilder.java

```diff
@@ -0,0 +1,215 @@
+ * builder.setInt64Value(1); MonitoringInfo mi = builder.build();
+ */
+public class SimpleMonitoringInfoBuilder {
+  public static final String ELEMENT_COUNT_URN =
+      BeamUrns.getUrn(MonitoringInfoUrns.Enum.ELEMENT_COUNT);
+  public static final String USER_COUNTER_URN_PREFIX =
+      BeamUrns.getUrn(MonitoringInfoUrns.Enum.USER_COUNTER_URN_PREFIX);
+  public static final String SUM_INT64_TYPE_URN =
+      BeamUrns.getUrn(MonitoringInfoTypeUrns.Enum.SUM_INT64_TYPE);
+
+  private static final HashMap<String, MonitoringInfoSpec> specs =
+      new HashMap<String, MonitoringInfoSpec>();
+
+  private final boolean validateAndDropInvalid;
+
+  private static final Logger LOG = LoggerFactory.getLogger(SimpleMonitoringInfoBuilder.class);
+
+  private MonitoringInfo.Builder builder;
+
+  static {
+    for (MonitoringInfoSpecs.Enum val : MonitoringInfoSpecs.Enum.values()) {
+      // Ignore the UNRECOGNIZED = -1 value;
+      if (!((Enum) val).name().equals("UNRECOGNIZED")) {
+        MonitoringInfoSpec spec =
+            val.getValueDescriptor().getOptions().getExtension(BeamFnApi.monitoringInfoSpec);
+        SimpleMonitoringInfoBuilder.specs.put(spec.getUrn(), spec);
+      }
+    }
+  }
+
+  public SimpleMonitoringInfoBuilder() {
+    this(true);
+  }
+
+  public SimpleMonitoringInfoBuilder(boolean validateAndDropInvalid) {
+    this.builder = MonitoringInfo.newBuilder();
+    this.validateAndDropInvalid = validateAndDropInvalid;
+  }
+
+  /** @return True if the MonitoringInfo has valid fields set, matching the spec */
+  private boolean validate() {
+    String urn = this.builder.getUrn();
+    if (urn == null || urn.isEmpty()) {
+      LOG.warn("Dropping MonitoringInfo since no URN was specified.");
+      return false;
+    }
+
+    MonitoringInfoSpec spec;
+    // If it's a user counter, and it has this prefix.
+    if (urn.startsWith(USER_COUNTER_URN_PREFIX)) {
+      spec = SimpleMonitoringInfoBuilder.specs.get(USER_COUNTER_URN_PREFIX);
+      List<String> split = Splitter.on(':').splitToList(urn);
+      if (split.size() != 4) {
+        LOG.warn(
+            "Dropping MonitoringInfo for URN %s, UserMetric namespaces and "
+                + "name cannot contain ':' characters.",
+            urn);
+        return false;
+      }
+    } else if (!SimpleMonitoringInfoBuilder.specs.containsKey(urn)) {
+      // Succeed for unknown URNs, this is an extensible metric.
+      // TODO(ajamato): Allow adding your own spec file to validate your metrics.
+      return true;
+    } else {
+      spec = SimpleMonitoringInfoBuilder.specs.get(urn);
+    }
+
+    if (!this.builder.getType().equals(spec.getTypeUrn())) {
+      LOG.warn(
+          "Dropping MonitoringInfo since for URN %s with invalid type field. Expected: %s"
+              + " Actual: %s",
+          this.builder.getUrn(), spec.getTypeUrn(), this.builder.getType());
+      return false;
+    }
+
+    Set<String> requiredLabels = new HashSet<String>(spec.getRequiredLabelsList());
```

Comment:
We can also add validation of different labels values. Such as PTransform should be a valid pTransform name, etc.